pkg/adapter/v2/main_message.go

```diff
@@ -27,19 +27,19 @@ import (
 	"go.opencensus.io/stats/view"
 	"go.uber.org/zap"
 	"knative.dev/eventing/pkg/metrics/source"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
 	"knative.dev/pkg/logging"
 	"knative.dev/pkg/metrics"
 	"knative.dev/pkg/profiling"
 	"knative.dev/pkg/signals"
-
-	"knative.dev/eventing/pkg/kncloudevents"
 )
 
 type MessageAdapter interface {
 	Start(ctx context.Context) error
 }
 
-type MessageAdapterConstructor func(ctx context.Context, env EnvConfigAccessor, adapter *kncloudevents.HTTPMessageSender, reporter source.StatsReporter) MessageAdapter
+type MessageAdapterConstructor func(ctx context.Context, env EnvConfigAccessor, sink duckv1.Addressable, reporter source.StatsReporter) MessageAdapter
```

Comment:
Why is this API change necessary?

Reply:
We don't use the `HTTPMessageSender` anymore. Now we use the `CloudEventsRequest`, whichs `Send` method requires an addressable